@uppy/angular is not correctly packaged..

it is package with a more package.json in the root and then with a "dist" folder
that dist folder has then an angular folder that the actual angular package is in..

If i use a component that uses @uppy/angular and i compile that component with Angular 13 (or 12 in Ivy partial mode)

And i install that component in a project, also make sure @uppy/angular is installed (i see it in the node_modules/@uppy/angular)

then still when i build the project i get:
/node_modules/@servoy/servoyextracomponents/fesm2020/servoy-servoyextracomponents.mjs:26:0-38 - Error: Module not found: Error: Can't resolve '@uppy/angular' in '<some-path>/node_modules/@servoy/servoyextracomponents/fesm2020'

But it should be there right?
But typescript/angular compiler can't find it..

When i inspected the package is did see that indirection into the real one
If i now copy everything that is in node_modules/@uppy/angular/dist/angular to  node_modules/@uppy/angular/ (overwriting the package.json) so the dist/angular becomes the real root of the package
Everything starts to work and it compiles fine..

So is there a reason why there is a dist folder? and not directly just the actual dist contents that angular builds?

quickly tried it, checked out the uppy
go to the examples replaced all the uppy imports (that with workspace as version) to:

 "@uppy/angular": "^0.2.6",
    "@uppy/core": "^2.1.3",
    "@uppy/drag-drop": "^2.0.5",
    "@uppy/google-drive": "^2.0.5",
    "@uppy/progress-bar": "^2.0.4",
    "@uppy/tus": "^2.1.2",
    "@uppy/webcam": "^2.0.5",

then did npm run build:

./src/app/app.component.ts:7:0-36 - Error: Module not found: Error: Can't resolve '@uppy/angular' in 'D:\temp\uppytest\uppy\examples\angular-example\src\app'

and i get a lot of css errors also because of some parents references that i didn't,
but the main thing is that you can see @uppy/angular can't be found because the package of 0.2.6 is just broken..

if i then make it:

    "@uppy/angular": "0.2.5",

so i force 0.2.5 version
then i see (which i didn't see before)

⠙ Generating browser application bundles (phase: setup)...Processing legacy "View Engine" libraries:
- @uppy/angular [module/esm2015] ()

and no error that that module is not found..

--- a/packages/@uppy/angular/package.json
+++ b/packages/@uppy/angular/package.json
@@ -1,7 +1,7 @@
 {
   "name": "@uppy/angular",
   "version": "0.2.7",
-  "module": "dist/angular/esm2015/public-api.js",
+  "module": "dist/angular/esm2020/public-api.mjs",
   "types": "dist/angular/uppy-angular.d.ts",
   "scripts": {
     "ng": "ng",

